lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -234,6 +234,9 @@ static void addMandatoryDiagnosticOptPipeline(SILPassPipelinePlan &P) {
   // occur before IRGen.
   P.addMoveOnlyTypeEliminator();
 
+  // Needed to fold MemoryLayout constants in performance-annotated functions.
+  P.addTargetConstantFolding();
```

Comment:
That shouldn't be needed anymore.
You should remove the whole TargetConstantFolding pass in this PR.

Reply:
Done